network/p2p/subscription_filter.go

```diff
@@ -0,0 +1,79 @@
+package p2p
+
+import (
+	"strings"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/libp2p/go-libp2p-pubsub/pb"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/network"
+)
+
+type Filter struct {
```

Comment:
So this is not a generic filter, that allows a customizable policy, right? It's a specific one, that tries to guess what to allow a remote node to subscribe to based on their role.

Would it be appropriate to rename this to e.g. `RoleBasedFilter` and leave a code comment stating the intent of the policy enforced here?